### DISCUSSION TITLE
Search whole array for any match

### DISCUSSION THREAD

**Discussion by philipainsaar** [supabase/supabase#2050] [Questions] [answered]
Is there any way to do something similar to this in the javascript api?

`select * from groups where 'ce3acb8c-945e-49fe-bcf3-514143bbc9c4'=ANY(has_access);`

I have an UUID array column in my table and would like to see if there is a match at any array index. 

Thankful for any help or guidance

**Comment by soedirgo** [comment, marked as answer]
Hey @philipainsaar, can you try the following?
```js
await supabase.from('groups').select().contains('has_access', ['ce3acb8c-945e-49fe-bcf3-514143bbc9c4'])
```

**Reply by philipainsaar** [reply]
Hey @soedirgo wow that worked, thank you! 

Is the ".cs" filter similar to ".contains"? I had tried the ".contains" filter here before without any luck
[](https://supabase.io/docs/reference/javascript/contains)

**Reply by soedirgo** [reply]
Oops, you're right, sorry! `contains` is the preferred method, `cs` was deprecated in favor of it. I guess you may have passed the UUID as an argument instead of a 1-member list, or maybe something else, but `contains` and `cs` use the same code.

**Reply by roker15** [reply]
> Hey @philipainsaar, can you try the following?
> 
> ```js
> await supabase.from('groups').select().contains('has_access', ['ce3acb8c-945e-49fe-bcf3-514143bbc9c4'])
> ```

What if i pass empty array? Does it return all or none?

**Reply by GaryAustin1** [reply]
@roker15 
An empty array will return all rows with any array (empty or with items).  It will not return rows with null for the array column.

**Reply by Rc85** [reply]
This will return rows where has_access is empty. Is there a method that actually uses the `ANY` function? `.contains` uses the `in` operator.

**Reply by steve-chavez** [reply]
@Rc85 It's possible to use `ANY/ALL` on the REST API: https://postgrest.org/en/stable/references/api/tables_views.html#operator-modifiers

On `supabase-js` there are some any methods exposed: https://github.com/supabase/postgrest-js/blob/master/test/filters.ts#L205-L209
